### ISSUE TITLE
Option AlwaysMultipartFormData with AddParamter(name, value contentType, ParameterType.RequestBody) without file breaks the request

### ISSUE THREAD

**Issue by offtopic** [restsharp/RestSharp#1134]
`request.AddParameter` with `ContentType` and `ParameterType.RequestBody` in conjunction with `AlwaysMultipartFormData = true` is not working, if no file is added.
The multipart body is empty

## Expected Behavior
```
POST http://dev.null/create HTTP/1.1
Accept: application/json, application/xml, text/json, text/x-json, text/javascript, text/xml
User-Agent: RestSharp/106.3.1.0
Content-Type: multipart/form-data; boundary=-----------------------------28947758029299
Host: dev.null
Content-Length: 363
Accept-Encoding: gzip, deflate
Proxy-Connection: Keep-Alive

-------------------------------28947758029299
Content-Type: application/json
Content-Disposition: form-data; name="metadata"

{"Prop1":"abc","Prop2":"efg"}
-------------------------------28947758029299--
```

## Actual Behavior
```
POST http://dev.null/create HTTP/1.1
Accept: application/json, application/xml, text/json, text/x-json, text/javascript, text/xml
User-Agent: RestSharp/106.3.1.0
Content-Type: multipart/form-data; boundary=-----------------------------28947758029299
Host: dev.null
Content-Length: 49
Accept-Encoding: gzip, deflate
Proxy-Connection: Keep-Alive

-------------------------------28947758029299--
```

## Steps to Reproduce the Problem
```
using RestSharp;

namespace ConsoleApp1
{
    class Program
    {
        static void Main(string[] args)
        {
            var client = new RestClient("http://dev.null");
            var request = new RestRequest("create")
            {
                AlwaysMultipartFormData = true,
                RequestFormat = DataFormat.Json,
                Method = Method.POST
            };

            request.AddFile("binary",  @"c:\temp\2.txt", "text/plain"); // if added it works as expected
            request.AddParameter("metadata", request.JsonSerializer.Serialize(new { Prop1 = "abc", Prop2 = "efg" }), "application/json", ParameterType.RequestBody);

            // if used without content type and parameterType.RequestBody it is correctly added, but content type is missing
            //request.AddParameter("metadata1", request.JsonSerializer.Serialize(new { Prop1 = "abc", Prop2 = "efg" }));
            client.Execute(request);
        }
    }
}
```

## Specifications

  - Version: Restsharp 106.3.1 (via nuget)
  - Platform: Console App .Net Framework 4.6.1

**Comment by offtopic** [comment]
I have made a fork and create a fix - i can also create a pull request:
https://github.com/marmind/RestSharp/commit/4cfe8c6aca21546eb5ff4a6484e0e26e3af235ae

**Comment by offtopic** [comment]
Unfortunately my "fix" breaks another valid use-case, I have to investigate more (but no time ;( )

**Comment by offtopic** [comment]
The fix works ...
Pull request  #1135 with new test `MultipartFormData_Without_File_Creates_A_Valid_RequestBody`
created.

**Comment by alexeyzimarev** [comment]
Fixed by #1135
